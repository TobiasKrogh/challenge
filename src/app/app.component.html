<header>
  <challenge-header></challenge-header>
</header>

<main>
  <challenge-navigation></challenge-navigation>
  <router-outlet></router-outlet>
</main>

<footer>
  <challenge-footer></challenge-footer>
</footer>
