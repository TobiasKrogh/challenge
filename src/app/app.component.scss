:host {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

header {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

main {
  display: flex;
  flex: 1;
  padding-right: var(--space-right);
}

footer {
  align-items: center;
  display: flex;
  justify-content: space-between;
}
