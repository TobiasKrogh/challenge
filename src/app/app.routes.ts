import { Route } from '@angular/router';
import { NotFoundComponent } from './components/not-found/not-found.component';

export const appRoutes: Route[] = [
  {
    path: '',
    pathMatch: 'full',
    redirectTo: '/reports',
  },
  {
    path: 'reports',
    loadChildren: () =>
      import('@challenge/reports').then((m) => m.ReportsModule),
  },
  {
    path: '**',
    component: NotFoundComponent,
  },
];
