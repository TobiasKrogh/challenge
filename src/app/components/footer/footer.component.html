<a routerLink="/tac">Terms&Conditions | Privacy policy</a>
