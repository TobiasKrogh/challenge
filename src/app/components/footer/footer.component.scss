:host {
  align-items: center;
  display: flex;
  height: 3.75rem;
  padding-left: var(--navigation-width);
}
