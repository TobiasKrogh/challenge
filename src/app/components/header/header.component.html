<div class="logo">
  <!-- <a href="#"> -->
  <img src="/assets/images/logo.v1.svg" alt="Logo" />
  <!-- </a> -->
</div>

<div class="menu">
  <button class="menu-button" type="button">
    <img src="/assets/images/menu.v1.svg" alt="Menu" />
  </button>
</div>

<ng-container *ngIf="user$ | async as user">
  <challenge-user [user]="user"></challenge-user>
</ng-container>
