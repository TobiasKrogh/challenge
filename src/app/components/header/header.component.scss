:host {
  align-items: center;
  border-bottom: 0.125rem solid var(--hex-color-grey--light);
  display: flex;
  flex: 1;
  height: 5.25rem;
  padding-right: var(--space-right);
}

.logo {
  display: flex;
  justify-content: center;
  width: var(--navigation-width);
}

.menu {
  align-items: center;
  display: flex;
  flex: 1;

  &-button {
    height: 2.6875rem;
    margin-left: -0.375rem;
  }
}
