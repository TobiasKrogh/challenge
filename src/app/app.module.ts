import { HttpClient, HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { TranslateLoader, TranslateModule } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { API_BASE_PATH_TOKEN } from '@challenge/api';
import { MessageComponent } from '@challenge/shared';
import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { appRoutes } from './app.routes';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { UserComponent } from './components/header/user/user.component';
import { NavigationComponent } from './components/navigation/navigation.component';
import { NotFoundComponent } from './components/not-found/not-found.component';

export function createTranslateLoader(http: HttpClient) {
  return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
    NavigationComponent,
    UserComponent,
    NotFoundComponent,
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    RouterModule.forRoot(appRoutes, { initialNavigation: 'enabledBlocking' }),
    TranslateModule.forRoot({
      defaultLanguage: 'en',
      loader: {
        provide: TranslateLoader,
        useFactory: createTranslateLoader,
        deps: [HttpClient],
      },
    }),
    MessageComponent,
  ],
  providers: [
    {
      provide: API_BASE_PATH_TOKEN,
      useValue: environment.apiBasePath,
    },
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
