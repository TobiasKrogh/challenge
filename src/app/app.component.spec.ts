import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateService } from '@ngx-translate/core';
import { MockComponents } from 'ng-mocks';
import { AppComponent } from './app.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { NavigationComponent } from './components/navigation/navigation.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  let translateService: TranslateService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      providers: [
        {
          provide: TranslateService,
          useValue: {
            setDefaultLang: jest.fn(),
            use: jest.fn(),
          },
        },
      ],
      declarations: [
        MockComponents(HeaderComponent, NavigationComponent, FooterComponent),
        AppComponent,
      ],
    }).compileComponents();

    translateService = TestBed.inject(TranslateService);

    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it sets up language defaults', () => {
      fixture.detectChanges();
      expect(translateService.setDefaultLang).toHaveBeenCalledWith('en');
      expect(translateService.use).toHaveBeenCalledWith('en');
    });
  });
});
