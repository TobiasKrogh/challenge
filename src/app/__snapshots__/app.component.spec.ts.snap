// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent When rendering Then it matches snapshot 1`] = `
DebugElement {
  "nativeNode": <div
    id="root1"
  >
    <header>
      <challenge-header />
    </header>
    <main>
      <challenge-navigation />
      <router-outlet />
      
    </main>
    <footer>
      <challenge-footer />
    </footer>
  </div>,
}
`;
